### PR TITLE
[new release] resp-client, resp, resp-unix, resp-mirage and resp-server (0.10.0)

### PR DESCRIPTION
**Pull request by zshipko**
Redis serialization protocol client library

- Project page: <a href="https://github.com/zshipko/resp">https://github.com/zshipko/resp</a>
- Documentation: <a href="https://zshipko.github.io/resp">https://zshipko.github.io/resp</a>

##### CHANGES:

- Updated conduit and alcotest
- Cleanup `Resp.t` type

### CODE REVIEW

**Comment by zshipko** [comment]
I'm not sure what the issue with the `revdeps` is, but I'm happy to fix it!

File: packages/resp-client/resp-client.0.10.0/opam [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+doc: "https://zshipko.github.io/resp"
+license: "ISC"
+dev-repo: "git+https://github.com/zshipko/resp.git"
+bug-reports: "https://github.com/zshipko/resp/issues"
+tags: ["redis" "protocol"]
+
+depends:
+[
+    "ocaml" {>= "4.07.0"}
+    "dune" {build & >= "2.0.0"}
```
Comment (mseri) [inline packages/resp-client/resp-client.0.10.0/opam:14]:
```suggestion
    "dune" {>= "2.0.0"}
```

File: packages/resp-mirage/resp-mirage.0.10.0/opam [hunk 2]
```diff
@@ -0,0 +1,37 @@
+opam-version: "2.0"
+maintainer: "Zach Shipko <zachshipko@gmail.com>"
+authors: "Zach Shipko <zachshipko@gmail.com>"
+license: "ISC"
+homepage: "https://github.com/zshipko/resp"
+doc: "https://zshipko.github.io/resp"
+bug-reports: "https://github.com/zshipko/resp/issues"
+depends: [
+  "ocaml" {>= "4.07.0"}
+  "dune" {build & >= "2.0.0"}
```
Comment (mseri) [inline packages/resp-mirage/resp-mirage.0.10.0/opam:10]:
```suggestion
  "dune" {>= "2.0.0"}
```

File: packages/resp-server/resp-server.0.10.0/opam [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+authors: ["Zach Shipko <zachshipko@gmail.com>"]
+homepage: "https://github.com/zshipko/resp"
+doc: "https://zshipko.github.io/resp"
+license: "ISC"
+dev-repo: "git+https://github.com/zshipko/resp.git"
+bug-reports: "https://github.com/zshipko/resp/issues"
+depends:
+[
+    "ocaml" {>= "4.07.0"}
+    "dune" {build & > "2.0.0"}
```
Comment (mseri) [inline packages/resp-server/resp-server.0.10.0/opam:12]:
```suggestion
    "dune" {>= "2.0.0"}
```

File: packages/resp-unix/resp-unix.0.10.0/opam [hunk 4]
```diff
@@ -0,0 +1,41 @@
+opam-version: "2.0"
+maintainer: "Zach Shipko <zachshipko@gmail.com>"
+authors: "Zach Shipko <zachshipko@gmail.com>"
+license: "ISC"
+homepage: "https://github.com/zshipko/resp"
+doc: "https://zshipko.github.io/resp"
+bug-reports: "https://github.com/zshipko/resp/issues"
+depends: [
+  "ocaml" {>= "4.07.0"}
+  "dune" {build & >= "2.0.0"}
```
Comment (mseri) [inline packages/resp-unix/resp-unix.0.10.0/opam:10]:
```suggestion
  "dune" {>= "2.0.0"}
```

File: packages/resp/resp.0.10.0/opam [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+authors: ["Zach Shipko <zachshipko@gmail.com>"]
+homepage: "https://github.com/zshipko/resp"
+doc: "https://zshipko.github.io/resp"
+license: "ISC"
+dev-repo: "git+https://github.com/zshipko/resp.git"
+bug-reports: "https://github.com/zshipko/resp/issues"
+depends:
+[
+    "ocaml" {>= "4.07.0"}
+    "dune" {build & >= "2.0.0"}
```
Comment (mseri) [inline packages/resp/resp.0.10.0/opam:12]:
```suggestion
    "dune" {>= "2.0.0"}
```

Comment (mseri) [reply packages/resp/resp.0.10.0/opam:12]: Was the build flag also in the previous releases? If so,  that needs to be fixed (not necessarily here, can be a separate PR)

File: packages/resp-client/resp-client.0.10.0/opam [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    "resp" {= version}
```
Comment (mseri) [inline packages/resp-client/resp-client.0.10.0/opam:15]: Are the packages in 0.9 also requiring the strict version like here?
```
    "resp" {= version}
```

if not, we should probably modify them to do it

Comment (mseri) [reply packages/resp-client/resp-client.0.10.0/opam:15]: Looks like they were less strict: https://github.com/ocaml/opam-repository/blob/e248948afd8b084309f2db5165d753f2b056fc16/packages/resp-mirage/resp-mirage.0.9.1/opam#L13

they do need an upper bound though

File: packages/resp/resp.0.10.0/opam (hunk 5 above)
Comment (mseri) [reply packages/resp/resp.0.10.0/opam:12]: The flag was removed already in the previous releases

**Comment by zshipko** [comment]
Thanks! I’ve added those additional constraints

**Comment by mseri** [comment]
Thank you very much! Looks fine now

**Pull request merged**